terraform/groups/environments/bucket_path.tf

```diff
@@ -0,0 +1,5 @@
+resource "aws_s3_object" "environment" {
+  bucket     = local.signed_bucket_id
+  content_type = "application/x-directory"
+  key        = "${var.environment}/"
+}
```

Comment:
There don't appear to be any usages of this resource (i.e. `aws_s3_object.environment`) in the codebase. Presumably this was added to create the illusion of a directory named `${var.environment}/`, but this will result in the creation of a zero byte file of the same name, which could lead to confusion since there is no requirement to first create a directory (or folder) before storing objects in S3.

I'd suggest we remove this and instead rely on AWS to abstract [shared object prefixes](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-folders.html) as 'folders':

```suggestion
```

Reply:
Precisely that. For the most part developers won't be aware of the difference. This is purely to guide any configuration created that is most likely gleaned from the console and interpreted in a hierarchical fashion. This is now a moot point given our later discussion. I'll push further changes...